internal/pubkey/encryption_types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Finobo
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pubkey
+
+import (
+	"fmt"
+
+	"github.com/mailchain/mailchain/internal/protocols"
+)
+
+const (
+	// AES256CBC encryption type name.
+	AES256CBC = "aes256cbc"
+)
+
+// EncryptionMethods returns supported encryption methods.
+func EncryptionMethods(protocol string) ([]string, error) {
```

Comment:
Encryption method that is allow is defined by the `PublicKey.Kind()` value. E.g.
- secpk2561 = [aes256cbc, noop]
- ed25519 = [nacl, noop]

Protocols usually support one key type but substrate actually supports mulitple